frontend/src/pages/Scan.js

```diff
@@ -1,15 +1,108 @@
 import { Component } from "react";
 import { Pages } from "../globals/Enums";
+import QrScanner from "qr-scanner";
+
+import { toast } from "react-toastify";
+
+QrScanner.WORKER_PATH = "../qr-scanner-worker.min.js";
 
 export default class Scan extends Component {
     constructor(props) {
         super(props);
         this.state = {};
     }
+    qrScanner;
+
+    componentDidMount() {
+        const videoElem = document.getElementById("qrScan");
+        this.qrScanner = new QrScanner(videoElem, (result) =>
+            console.log("decoded qr code:", result)
+        );
+        this.qrScanner.start();
+    }
+    componentWillUnmount() {
+        this.qrScanner.stop();
+    }
+
+    async processQrCodeScan(qrCode) {
+        console.log(qrCode);
+
+        try {
+            const fields = qrCode.split(",");
+            const userId = fields[0];
+            const token = fields[1];
+            const name = fields[2];
+            const email = fields[3];
+            const phone = fields[4];
```

Comment:
Will this throw an exception if the name, email, phone are not provided?

Reply:
if my implementation of the QR code was good, potentially, but right now I'm putting `null` in the QR code if the value doesn't exist, so nothing explodes

Reply:
ive been testing with just userid and token so it SHOULD only break with a malformed qr code, in which case throwing the exception is desired